### PR TITLE
DOC Add versionadded and versionchanged to API reference

### PR DESCRIPTION
**Pull request by matthew-so**
Just a test to ensure that I am properly adding versionadded/versionchanged documentation. Are my current additions correct, given the functions' edit history?

### CODE REVIEW

File: fairlearn/datasets/_fetch_adult.py [hunk 1]
```diff
@@ -25,6 +25,9 @@ def fetch_adult(*, cache=True, data_home=None,
     Prediction task is to determine whether a person makes over $50,000 a
     year.
 
+    ..versionadded: 0.5.0
+    ..versionchanged: 0.7.1
```
Comment (adrinjalali) [inline fairlearn/datasets/_fetch_adult.py:29]: version changed should have a comment explaining what's changed.

Comment (romanlutz) [reply fairlearn/datasets/_fetch_adult.py:29]: In this particular case nothing changed other than the documentation, so I think we can remove that line and keep the `versionadded` one only.

Comment (romanlutz) [reply fairlearn/datasets/_fetch_adult.py:29]: Come to think of it, the same applies to the other two in this module.

**Comment by matthew-so** [comment]
Sorry for the delay; classes just started. I'll get it done as soon as possible.

**Review by romanlutz — APPROVED** [review]

**Comment by romanlutz** [comment]
@fairlearn/fairlearn-maintainers can we merge this?

**Comment by hildeweerts** [comment]
Did someone check whether this works locally? It seems that other libraries (e.g., scikit-learn) usually use something like
`.. versionadded:: 0.5.0` rather than `..versionadded: 0.5.0`, see also [sphinx docs](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-versionadded).

File: fairlearn/datasets/_fetch_adult.py [hunk 2]
```diff
@@ -25,6 +25,8 @@ def fetch_adult(*, cache=True, data_home=None,
     Prediction task is to determine whether a person makes over $50,000 a
     year.
 
+    ..versionadded: 0.5.0
```
Comment (romanlutz) [inline fairlearn/datasets/_fetch_adult.py:28]:
```suggestion
    .. versionadded: 0.5.0
```

File: fairlearn/datasets/_fetch_bank_marketing.py [hunk 3]
```diff
@@ -30,6 +30,7 @@ def fetch_bank_marketing(*, cache=True, data_home=None,
     The classification goal is to predict if the client will subscribe a
     term deposit (variable y).
 
+    ..versionadded: 0.5.0
```
Comment (romanlutz) [inline fairlearn/datasets/_fetch_bank_marketing.py:33]:
```suggestion
    .. versionadded: 0.5.0
```

File: fairlearn/datasets/_fetch_boston.py [hunk 4]
```diff
@@ -55,6 +55,7 @@ def fetch_boston(*, cache=True, data_home=None,
     MEDV     Median value of owner-occupied homes in $1000's
     =======  ======================================================================
 
+    ..versionadded: 0.5.0
```
Comment (romanlutz) [inline fairlearn/datasets/_fetch_boston.py:58]:
```suggestion
    .. versionadded: 0.5.0
```

**Review by romanlutz — COMMENTED** [review]
Thanks for the suggestion, @hildeweerts ! Better to be safe and add the whitespace.

**Comment by hildeweerts** [comment]
> Thanks for the suggestion, @hildeweerts ! Better to be safe and add the whitespace.

It's not just whitespace but also a double colon!

**Comment by romanlutz** [comment]
> > Thanks for the suggestion, @hildeweerts ! Better to be safe and add the whitespace.
> 
> It's not just whitespace but also a double colon!

Ooops. Updated!

**Comment by romanlutz** [comment]
@hildeweerts Thanks again for the suggestions!!! Do you want to merge this?

**Pull request merged**